src/tokens/Datatoken.ts

```diff
@@ -1357,4 +1363,105 @@ export class Datatoken {
     const balance = await dtContract.methods.balanceOf(address).call()
     return this.web3.utils.fromWei(balance)
   }
+
+  /**
+   * @dev estGasSetPublishingMarketFee
+   *      Estimating gas for publishMarketFeeAddress method
+   * @param {string} datatokenAddress Datatoken adress
+   * @param {string} publishMarketFeeAddress  new publish Market Fee Address
+   * @param {string} publishMarketFeeToken new publish Market Fee Token
+   * @param {string} publishMarketFeeAmount new fee amount
+   * @param {String} address user adress
+   */
+  public async estGasSetPublishingMarketFee(
+    datatokenAddress: string,
+    publishMarketFeeAddress: string,
+    publishMarketFeeToken: string,
+    publishMarketFeeAmount: string,
+    address: string
+  ): Promise<number> {
+    // Estimate gas cost for publishMarketFeeAddress method
+    const gasLimitDefault = this.GASLIMIT_DEFAULT
+    const dtContract = new this.web3.eth.Contract(this.datatokensAbi, datatokenAddress, {
+      from: address
+    })
+    let estGas
+    try {
+      estGas = await dtContract.methods
+        .setPublishingMarketFee(
+          publishMarketFeeAddress,
+          publishMarketFeeToken,
+          publishMarketFeeAmount
+        )
+        .estimateGas({ from: address }, (err, estGas) => (err ? gasLimitDefault : estGas))
+    } catch (error) {
+      estGas = gasLimitDefault
+    }
+    return estGas
+  }
+
+  /**
+   * @dev setPublishingMarketFee
+   *      Only publishMarketFeeAddress can call it
+   *      This function allows to set the fee required by the publisherMarket
+   * @param {string} datatokenAddress Datatoken adress
+   * @param {string} publishMarketFeeAddress  new publish Market Fee Address
+   * @param {string} publishMarketFeeToken new publish Market Fee Token
+   * @param {string} publishMarketFeeAmount new fee amount
+   * @param {String} address user adress
+   */
+  public async setPublishingMarketFee(
+    datatokenAddress: string,
+    publishMarketFeeAddress: string,
+    publishMarketFeeToken: string,
+    publishMarketFeeAmount: string,
+    address: string
+  ) {
+    const dtContract = new this.web3.eth.Contract(this.datatokensAbi, datatokenAddress, {
+      from: address
+    })
+    const estGas = await this.estGasSetPublishingMarketFee(
```

Comment:
it would be nice to check if the caller has the `publishMarketFeeAddress` role and throw an error if that's not the case. Not 100% sure that is possible, think we did that for `deployERC20` role for some methods

Reply:
The smart contract already checks this but we could add an additional check here too